src/mattack_actors.cpp

```diff
@@ -847,20 +872,43 @@ bool gun_actor::call( monster &z ) const
             }
             return false;
         }
-
+        aim_at = target->pos();
     } else {
         target = z.attack_target();
         if( !target || !z.sees( *target ) || ( !target->is_monster() && !z.aggro_character ) ) {
-            return false;
+            //return false;
+            if( !target_moving_vehicles ) {
+                return false;
+            }
+            //No living targets, try to find a moving car
+            vehicle *veh = find_target_vehicle( z, get_max_range() );
+            if( !veh ) {
+                return false;
+            }
+            for( const vpart_reference &vp : veh->get_avail_parts( "CONTROLS" ) ) {
+                if( z.sees( vp.pos() ) ) {
+                    aim_at = vp.pos();
+                    untargeted = true;
+                }
+            }
```

Comment:
this chunk feels sorta weird, unless i missed some checks (it is kinda late for me) this will turn monsters into clairvoyant snipers;
detecting moving vehicle doesn't seem to require sight or noise - monsters will shoot up to their max range anyway
for some reason they'll aim at controls - even migo/yrax or acid llama will know what CONTROLS look like
and only the first CONTROLS tile, creating a meme/meta for sacrificial set of armored controls

Reply:
I'm going to be honest here I am a cargo cultist so if there is a way to improve this, someone other than me will have to make the changes.  I agree with you it would make more sense for them to target randomly, wheels, engines, controls, everything else?  How would I go about making that happen?

Reply:
One way would be to grab vehicle->get_points() and roll via random_entry() from those points - that'll hit some random vehicle tile, or may be a more complex roll that weighs the front tiles more

a more weird thing is the lack of sight or noise checks, getting shot at by a feral or llama spitting acid at you while riding a silent bicycle in pitch dark seems odd, but may be i've missed the check for that or it happens earlier, or BN players don't care/complain loud enough

or just put it in as-is and try to balance/fix later when some gameplay testing occurs

Reply:
I would expect a (functioning) human to aim at the driver or where the driver ought to be if not directly visible (i.e. at the controls), unless trying to take out the vehicle specifically by firing an RPG at the engine block or trying to take out the wheels. On a normal car you know where the driver seat is located, while the Cataclysm custom jobs may require some more intelligence (which the game doesn't have unless you use the controls as a proxy) to figure out where the driver might be located.

A non intelligent opponent would probably fire at the center, as perceived from their perspective, i.e. the center tile on the facing "side" (easy for orthogonal orientations, but trickier for the rest), with dispersion from that location hitting tiles further away to either side.

Reply:
Playtested it a bit;
Spawned m2hb turret 60 tiles away - tested both daylight and new moon (no ambient light), the turret had no vision indicator in either but; It will happily frolick while player is in control of velocity=0 bicycle, the moment bicycle starts moving it will shoot to beyond it's daylight sight range, the same happens with migo, acid llamas etc

So no vision or noise checks made means they'll detect moving vehicles in reality bubble - also behind several walls / smoke etc

Reply:
So then it definitely needs more work. I’ll leave it open for people to make suggestions and commit them but otherwise this probably isn’t mergeable.